src/sagemaker/workflow/utilities.py

```diff
@@ -151,3 +159,41 @@ def validate_step_args_input(
         raise TypeError(error_message)
     if step_args.caller_name not in expected_caller:
         raise ValueError(error_message)
+
+
+def override_pipeline_parameter_var(func):
+    """A decorator to override pipeline Parameters passed into a function
+
+    This is a temporary decorator to override pipeline Parameter objects with their default value
+    and display warning information to instruct users to update their code.
+
+    This decorator can help to give a grace period for users to update their code when
+    we make changes to explicitly prevent passing any pipeline variables to a function.
+
+    We should remove this decorator after the grace period.
+    """
+    warning_msg_template = (
+        "%s should not be a pipeline variable (%s). "
+        "The default_value of this Parameter object will be used to override it. "
+        "Please remove this pipeline variable and use python primitives instead."
+    )
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        params = inspect.signature(func).parameters
```

Comment:
nit: Is there a reason for this? can't we enumerate over `args` directly?

Reply:
Yes, the reason is args only can not give us the name of the argument, so that we're not able to specify the invalid argument name in the warning message.
`Args` combined with `.signature` can provide us both the arg name and the arg value.

Reply:
ah, got you